src/sentry/api/endpoints/monitor_checkin_details.py

```diff
@@ -125,7 +125,8 @@ def put(self, request, project, monitor, checkin):
         with transaction.atomic():
             checkin.update(**params)
             if checkin.status == CheckInStatus.ERROR:
-                monitor.mark_failed(current_datetime)
+                if not monitor.mark_failed(current_datetime):
+                    return self.respond(serialize(checkin, request.user), status=208)
```

Comment:
@mnoble thoughts on using 208? in this case its implying "this is already marked as a failure, so your checkin didnt change the status"

Reply:
Technically I think you're only suppose to use 208 when a `DAV` header is present (for backwards compatibility, apparently, according to the spec: https://tools.ietf.org/html/rfc5842#page-31) Compatibility with what, I'm not sure.

I think the endpoint being idempotent is more important than the 200 or 208, but 208 sounds vaguely appropriate here.

Reply:
You could also use [205](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/205) to indicate that the client needs to reset their world view in order to make a proper request but not give an error.

Reply:
Well in this case it *is* making changes, its just not cascading all of those (in this case it means they duplicated a report or reported things out of order).

Reply:
Note that HTTP2 has special handling for known 200 codes but not for 208 or 205 when cache controls are used. Does not apply for post, but probably does indicate that one should not use unknown status codes to HTTP.

Also 205 cannot be used because 205 must not carry a body or it will break keep-alive connections.

Reply:
I didn't know about the special handling in HTTP2. 202 would be my next suggestion as it is a non-commital acceptance status code.

Reply:
or we turn it around. 201 in normal case and 200 on second attempt!

Reply:
201 makes sense for the checkin creation (gonna change) but not the checkin update

Reply:
im also ok going back to 200, just found it helpful to have the metadata